### ISSUE TITLE
Doc generation does not work with multipart lenses 

### ISSUE THREAD

**Issue by abs-zero** [http4k/http4k#323]
Here is a relevant snippet of what I did:

```kotlin
        val reqBody = Body.auto<RequestDto>().toLens()
        val jsonMultipart = MultipartFormField.required("json")
        val pictureMultipart = MultipartFormFile.required("picture")
        val picture2Multipart = MultipartFormFile.required("picture2")
        val extractMultipart = Body.multipartForm(Validator.Strict, spaceMultipart, picture2Multipart).toLens()

        return "/someRoute" meta {
            summary = "Add Spaces"
            receiving(extractMultipart)
            returning(Status.CREATED)
        } bindContract Method.POST to { req -> ... }
```

When I call this route with a valid request, I get
```json
{
    "message": "Missing/invalid parameters",
    "params": [
        {
            "name": "json",
            "type": "form",
            "datatype": "string",
            "required": true,
            "reason": "Invalid"
        },
        {
            "name": "picture",
            "type": "form",
            "datatype": "file",
            "required": true,
            "reason": "Invalid"
        }
    ]
}
```

If I omit `receving`, it works as expected with the same valid request. So the problem here is `receiving`.

Is there a different way to generate doc for multipart request? Or is this a bug?

**Comment by daviddenton** [comment]
There are 2 things here. 

1. The problem with the call is that the fields are extracted once by http4k as a part of the validation step for the contract checking, and then again when you try to use the value inside your handler. This is a problem for multipart because we implemented it by reading the inputstream to parse the request, and hence on the second time around the stream is empty and blows up.

You can work around this by telling http4k to not try and extract the body when validating:
```  "/bad-request-body-ignore-body" meta {
            receiving(requiredBody)
            preFlightExtraction = IgnoreBody
        } bindContract GET to { Response(OK) }```

2. The multipart fields do not actually render fully in the JSON output for the Swagger. This needs to be fixed.

**Comment by daviddenton** [comment]
This fix will be released in 3.206.0
